lara-typescript/src/example-interactives/src/attachments/runtime.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from "react";
+import { useState, useEffect } from "react";
+import { getAttachmentUrl, IRuntimeInitInteractive, writeAttachment, WriteAttachmentParams, IAttachmentInfo } from "../../../interactive-api-client";
+
+interface Props {
+  initMessage: IRuntimeInitInteractive<any, {}>;
+}
+
+interface UserAttachmentInfo extends IAttachmentInfo {
+  url?: string;
+  err?: any;
+}
+type UserAttachmentMap = Record<string, UserAttachmentInfo>;
+
+export const RuntimeComponent: React.FC<Props> = ({initMessage}) => {
+  const [uploading, setUploading] = useState<boolean>(false);
+  const [attachments, setAttachments] = useState<UserAttachmentMap>({});
+
+  // for now this is the best way to know in authoring runtime mode
+  const inAuthoringRuntimeMode = !initMessage.hostFeatures.getFirebaseJwt;
+
+  useEffect(() => {
+    const initialAttachments: UserAttachmentMap = {};
+    if (initMessage.attachments) {
+      Object.keys(initMessage.attachments).forEach(key => initialAttachments[key] = {});
+    }
+    setAttachments(initialAttachments);
+  }, [initMessage]);
+
+  useEffect(() => {
+    const getUrls = async () => {
+      const names = Object.keys(attachments);
+      for await (const name of names) {
+        const attachment = attachments[name];
+        if (!attachment.url) {
+          try {
+            const url = await getAttachmentUrl({name});
```

Comment:
One problem with this, is that the URLs expire. So if the user loads the page and waites past the experation date the file won't load unless the audio tag does pre caching.

Even if the audio tag does pre caching, it would be good to make that clear somehow in this example interactive since developer writing attachment interactives might refer to this example.

Reply:
I've updated the code to load the audio on a user click now - https://github.com/concord-consortium/lara/pull/913/commits/3a458ace5a4f39a0448502c5c95ac6ab1ffe5714